### PR TITLE
Use cached wallet_getCapabilities

### PR DESCRIPTION
**Pull request by KannuSingh**
Action wallet_getCapabilities will return cached data from session.sessioProperties if present else it will send a call to wallet to get the capabilities.

Added RequestLoaderModal.tsx for waiting for request to get fulfil from the wallet, this modal have different display message, RequestModal can be refactored to support both, but for simplicity created a separate modal to handle request which don't need a user action

### CODE REVIEW

File: advanced/dapps/react-dapp-v2/src/modals/RequestLoaderModal.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
```
Comment (arein) [inline advanced/dapps/react-dapp-v2/src/modals/RequestLoaderModal.tsx:1]: looks very similar to the old one - why a new one?

**Review by arein — COMMENTED** [review]
The `wallet_getCapabilities` "caching" should ultimately move into the provider. No concern with this PR but @KannuSingh can you leave a comment in the code reminding us to clean up after it's done.

@ganchoradkov the [ERC](https://eips.ethereum.org/EIPS/eip-5792) is in `review` now. When do you think is the right time to move this logic into the provider?

File: advanced/dapps/react-dapp-v2/src/modals/RequestLoaderModal.tsx (hunk 1 above)
Comment (KannuSingh) [reply advanced/dapps/react-dapp-v2/src/modals/RequestLoaderModal.tsx:1]: its just some text changes on the modal, it don't need any approval or reject action on wallet side.
I can refactor the old one to take these messages as props.

**Review by lukaisailovic — APPROVED** [review]

File: advanced/dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx [hunk 2]
```diff
@@ -548,8 +554,9 @@ export function JsonRpcContextProvider({
             `Missing rpcProvider definition for chainId: ${chainId}`
           );
         }
+        if(lastTxId == undefined) throw new Error(`Last transaction id  ${lastTxId}, make sure call to sendCalls returns successfully. `);
         //hardcoded valid userOpHash
```
Comment (bkrem) [inline advanced/dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:558]: No longer hardcoded based on the change (I presume)
```suggestion
```

File: advanced/dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx [hunk 3]
```diff
@@ -548,8 +554,9 @@ export function JsonRpcContextProvider({
             `Missing rpcProvider definition for chainId: ${chainId}`
           );
         }
+        if(lastTxId == undefined) throw new Error(`Last transaction id  ${lastTxId}, make sure call to sendCalls returns successfully. `);
```
Comment (bkrem) [inline advanced/dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:557]: Better to use `===` strict comparison here if the possible values are `string | undefined`

```suggestion
        if (lastTxId === undefined) throw new Error(`Last transaction ID is undefined, make sure previous call to sendCalls returns successfully. `);
```

**Review by bkrem — APPROVED** [review]

**Pull request merged**